pkg/webhook/elasticquota/quota_topology.go

```diff
@@ -0,0 +1,475 @@
+				quotaInfo.ParentName, quotaInfo.Name)
+		}
+	}
+
+	children, find := qt.quotaHierarchyInfo[quotaInfo.Name]
+	if !find || len(children) == 0 {
+		return nil
+	}
+
+	for name := range children {
+		if child, exist := qt.quotaInfoMap[name]; exist {
+			if !checkQuotaKeySame(quotaInfo.CalculateInfo.Max, child.CalculateInfo.Max) {
+				return fmt.Errorf("checkSubAndParentGroupMaxQuotaKeySame failed: %v's key is not the same with %v",
+					quotaInfo.Name, name)
+			}
+		} else {
+			return fmt.Errorf("BUG quotaInfoMap and quotaTree information out of sync, losed :%v", name)
+		}
+	}
+
+	return nil
+}
+
+// checkMinQuotaSum parent's minQuota should be larger or equal than the sum of allChildren's minQuota.
+func (qt *quotaTopology) checkMinQuotaSum(quotaInfo *core.QuotaInfo) error {
+	if quotaInfo.ParentName != extension.RootQuotaName {
+		childMinSumNotIncludeSelf, err := qt.getChildMinQuotaSumExceptSpecificChild(quotaInfo.ParentName, quotaInfo.Name)
+		if err != nil {
+			return fmt.Errorf("checkMinQuotaSum failed: %v", err)
+		}
+
+		childMinSumIncludeSelf := quotav1.Add(childMinSumNotIncludeSelf, quotaInfo.CalculateInfo.Min)
+		if isLessEqual, _ := quotav1.LessThanOrEqual(childMinSumIncludeSelf, qt.quotaInfoMap[quotaInfo.ParentName].CalculateInfo.Min); !isLessEqual {
+			return fmt.Errorf("checkMinQuotaSum allChildren SumMinQuota > parentMinQuota, parent: %v", quotaInfo.ParentName)
+		}
+	}
+
+	children, exist := qt.quotaHierarchyInfo[quotaInfo.Name]
+	if !exist || len(children) == 0 {
+		return nil
+	}
+
+	childMinSum, err := qt.getChildMinQuotaSumExceptSpecificChild(quotaInfo.Name, "")
+	if err != nil {
+		return fmt.Errorf("checkMinQuotaSum failed:%v", err)
+	}
+
+	if isLessEqual, _ := quotav1.LessThanOrEqual(childMinSum, quotaInfo.CalculateInfo.Min); !isLessEqual {
+		return fmt.Errorf("checkMinQuotaSum allChildrn SumMinQuota > MinQuota, parent: %v", quotaInfo.Name)
+	}
+
+	return nil
+}
+
+func (qt *quotaTopology) getChildMinQuotaSumExceptSpecificChild(parentName, skipQuota string) (allChildQuotaSum v1.ResourceList, err error) {
+	allChildQuotaSum = v1.ResourceList{}
+	if parentName == extension.RootQuotaName {
+		return allChildQuotaSum, nil
+	}
+
+	childQuotaList, exist := qt.quotaHierarchyInfo[parentName]
+	if !exist {
+		return nil, fmt.Errorf("not found child quota list, parent: %v", parentName)
+	}
+
+	for childName := range childQuotaList {
+		if childName == skipQuota {
+			continue
+		}
+
+		if quotaInfo, exist := qt.quotaInfoMap[childName]; exist {
+			allChildQuotaSum = quotav1.Add(allChildQuotaSum, quotaInfo.CalculateInfo.Min)
+		} else {
+			err = fmt.Errorf("BUG quotaInfoMap and quotaTree information out of sync, losed :%v", childName)
+			return nil, err
+		}
+	}
+
+	return allChildQuotaSum, nil
+}
+
+// checkMaxQuota maxQuota should be larger or equal than each child's maxQuota.
+func (qt *quotaTopology) checkMaxQuota(quotaInfo *core.QuotaInfo) error {
```

Comment:
there is no limit of childMaxSum must <= parentMax

Reply:
fixed